### DISCUSSION TITLE
Power Platform CLI: June Update Blog

### DISCUSSION THREAD

**Discussion by kkanakas** [microsoft/powerplatform-build-tools#429] [Announcements]
Here is the update for the new release of the Power Platform Command line: 
https://devblogs.microsoft.com/powerplatform/power-platform-command-line-june-update/

**Comment by Roodmack** [comment]
HI , 
I try the one with copilot feature to generate fetch xml querry to for dataverse, every time the xml is generate when i try to querry the Dataverse table i end up with this Error

**command :** 
---------------------------------------------------------------------------------------------------------------------------
pac copilot predict -m a1afa5d4-7a44-4c31-9cd2-e852a78431fa -p "list all data in account table  in fetchxml" > DVdemoEmployerFetch.xml
cat .\DVdemoEmployerFetch.xml
pac org fetch -xf .\DVdemoEmployerFetch.xml 
------------------------------------------------------------------------------------------------------------------------------
**Error :** 

------------------------------------------------------------------------------------------------------------------------------------------------
Error: 'Account' entity doesn't contain attribute with Name = 'opendeals' and NameMapping = 'Logical'. MetadataCacheDetails: ProviderType=Dynamic, StandardCache=True, IsLoadedInStagedContext = False, Timestamp=8588282, MinActiveRowVersion=8588282, MetadataInstanceId=8667247, LastUpdated=2023-07-25 07:04:39.357, OrgId=74aa606e-d327-4076-8360-a163bbebfc9e

---------------------------------------------------------------------------------------------------------------------------------------------------

**Reply by petrochuk** [reply]
@Roodmack there is one step missing:

1. Call GPT or get FetchXML from any other Large Language Model
`.\pac.exe copilot predict -n GPT -p "list all data in account table in fetchxml" > DVdemoEmployerFetch.xml`
2. FetchXML coming from LLMs is not matching your environment's schema. Run prepare-fetch to make sure FetchXML is matching it.
`.\pac.exe copilot prepare-fetch -i .\DVdemoEmployerFetch.xml -o .\DVdemoEmployerFetch_Ready.xml`
3. Execute FetchXML 
`.\pac.exe org fetch -xf .\DVdemoEmployerFetch_Ready.xml`

![image](https://github.com/microsoft/powerplatform-build-tools/assets/30735471/dd27025e-5ec8-4860-9438-d1f86917d365)

**Reply by rajyraman** [reply]
@petrochuk - What is -n param? I cannot see it in help.

![image](https://github.com/microsoft/powerplatform-build-tools/assets/5035266/9c5071bf-f661-47f2-aad4-affc79d0d30e)

**Reply by kkanakas** [reply]
It is a typo, it is -m

**Reply by petrochuk** [reply]
@rajyraman -n is not a typo. It is new parameter which will accept model name, so users don't have to pass model id.

**Reply by kkanakas** [reply]
New parameter being disclosed prior to release 😏@petrochuk

**Reply by petrochuk** [reply]
It is such a groundbreaking change to being able to pass name in addition to id ;)

**Reply by kkanakas** [reply]
🤣🤣
